components/application-broker/internal/broker/provision.go

```diff
@@ -394,6 +412,44 @@ func (svc *ProvisionService) channelForApp(applicationName internal.ApplicationN
 	return svc.knClient.GetChannelByLabels(integrationNamespace, labels)
 }
 
+// Create a new policy
+func (svc *ProvisionService) createIstioPolicy(ns internal.Namespace) error {
+	policyName := string(ns) + "-broker-filter"
+	svc.log.Infof("Creating Policy %s in namespace: %s", policyName, string(ns))
+
+	brokerTargetSelector := &istioauthenticationalpha1.TargetSelector{
+		Name: brokerTargetSelectorName,
+	}
+	filterTargetSelector := &istioauthenticationalpha1.TargetSelector{
+		Name: filterTargetSelectorName,
+	}
+	mtls := &istioauthenticationalpha1.MutualTls{
+		//PERMISSIVE
+		Mode: 1,
+	}
+	peerAuthenticationMethod := &istioauthenticationalpha1.PeerAuthenticationMethod{
+		Params: &istioauthenticationalpha1.PeerAuthenticationMethod_Mtls{
+			Mtls: mtls,
+		},
+	}
+
+	_, error := svc.istioClient.AuthenticationV1alpha1().Policies(string(ns)).Create(&istiov1alpha1.Policy{
```

Comment:
Would it make sense to scoop out a method to get a new Policy? That will make it more readable IMO.

Reply:
Well, I would say it is only 6 lines to be separated, plus this function would have a single client.